flake.nix

```
@@ -10,13 +10,35 @@
   outputs = { self, nixpkgs, nixlib }:
 
   # Library modules (depend on nixlib)
-  {
+  rec {
     # export all generator formats in ./formats
     nixosModules = nixlib.lib.mapAttrs' (file: _: {
       name = nixlib.lib.removeSuffix ".nix" file;
       # The exported module should include the internal format* options
       value.imports = [ (./formats + "/${file}") ./format-module.nix ];
     }) (builtins.readDir ./formats);
+
+    # example usage in flakes:
+    #   outputs = { self, nixpkgs, nixos-generators, ...}: {
+    #     vmware = nixos-generators.nixosGenerate {
+    #       pkgs = nixpkgs.legacyPackages.x86_64-linux;
+    #       modules = [./configuration.nix];
+    #       format = "vmware";
+    #   };
+    # }
+    nixosGenerate = { pkgs, modules, format }:
```

I think we can merge this after this nitpick:

```suggestion
    nixosGenerate = { pkgs, format,  modules ? [ ]}:
```